Source/WebCore/dom/StaticRange.cpp

```diff
@@ -83,4 +83,24 @@ void StaticRange::visitNodesConcurrently(JSC::AbstractSlotVisitor& visitor) cons
     addWebCoreOpaqueRoot(visitor, end.container.get());
 }
 
+bool StaticRange::computeValidity() const
+{
+    Node& startContainer = this->startContainer();
+    Node& endContainer = this->endContainer();
+
+    // Its start is before or equal to its end, if same container.
+    if (endOffset() < startOffset() && &startContainer == &endContainer)
```

Comment:
This check should probably appear immediately above is_gt since the relevant clause in the spec here is "Its [start](https://dom.spec.whatwg.org/#concept-range-start) is [before](https://dom.spec.whatwg.org/#concept-range-bp-before) or [equal](https://dom.spec.whatwg.org/#concept-range-bp-equal) to its [end](https://dom.spec.whatwg.org/#concept-range-end)."

By the way, these "what" comments are unnecessary. They just repeat what the code already conveys.

Reply:
Will make this change and take out the comments.